### PR TITLE
admin views and flows

### PR DESCRIPTION
**Pull request by rebekahkwon**

### CODE REVIEW

**Comment by rebekahkwon** [comment]
Bradley and I touched base 10pm. If you guys are okay, he's going to review, add routes, and merge for MVP.

**Comment by brianpchsu** [comment]
Sure.

**Comment by brianpchsu** [comment]
Thanks for doing this. I still need some time to catch up with react.

File: client/scripts/components/layouts/app.jsx [hunk 1]
```diff
@@ -6,16 +6,19 @@ var RouteHandler = Router.RouteHandler;
 
 var Header = require('../modules/Header.jsx');
 var Footer = require('../modules/Footer.jsx');
+var ElectionForm = require('../modules/ElectionForm.jsx');
+var ElectionAdmin = require('../modules/ElectionAdmin.jsx');
```
Comment (bportnoy) [inline client/scripts/components/layouts/app.jsx]: can you remove these extraneous requires?

File: client/scripts/components/layouts/app.jsx [hunk 2]
```diff
@@ -6,16 +6,19 @@ var RouteHandler = Router.RouteHandler;
...
+var PollForm = require('../modules/PollForm.jsx');
 
 var App = React.createClass({
 
-  // componentDidMount: function() {
-  //   //something will go here
-  // },
+  componentDidMount: function() {
+    //something will go here
```
Comment (bportnoy) [inline client/scripts/components/layouts/app.jsx]: and the component did mount / get initial state

File: client/scripts/components/modules/Answer.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+var React = require('react');
+var ReactPropTypes = React.PropTypes;
+var axios = require('axios');
```
Comment (bportnoy) [inline client/scripts/components/modules/Answer.jsx]: don't need axios or the prop types here

File: client/scripts/components/modules/ElectionForm.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        two_factor_auth:        ReactPropTypes.bool,
+        force_two_factor_auth:  ReactPropTypes.bool,
+        locked:                 ReactPropTypes.bool,
+        timed:                  ReactPropTypes.bool,
+        accepting_votes:        ReactPropTypes.bool
+    },
+
+    router: React.PropTypes.func,
+
+    handlesubmit: function(e) {
```
Comment (bportnoy) [inline client/scripts/components/modules/ElectionForm.jsx]: should be camelCase

File: client/scripts/components/modules/ElectionForm.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        e.preventDefault();
+        this.props.onElectionSubmit({
```
Comment (bportnoy) [inline client/scripts/components/modules/ElectionForm.jsx]: where is this function defined? and why are we trimming boolean values?

File: client/scripts/components/modules/PollForm.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+          <textarea rows='4' cols='70' className='description-field' placeholder='Enter a detailed description of the election, 1000 characters max' ref='description'></textarea>
+          <hr />
+
+          <div>{ questions }</div>
+
+          <button type='button' className='btn' onClick={ this.addAnotherQuestion }>ADD A QUESTION</button>
+          <br />
+
+          <a href='#' className='cancel'>CANCEL</a>
+          <input type='submit' action='Post' value='SAVE' />
```
Comment (bportnoy) [inline client/scripts/components/modules/PollForm.jsx]: this probably needs a submit function

File: client/scripts/routes.jsx [hunk 7]
```diff
@@ -8,15 +8,9 @@ var Route = Router.Route;
...
 var Index = require('./components/index.jsx');
 var App = require('./components/layouts/app.jsx');
-var Vote = require('./components/modules/vote/VotingBooth.jsx');
-var Submitting = require('./components/modules/vote/Submitting.jsx');
-var Submitted = require('./components/modules/vote/Submitted.jsx');
 
 var routes = (
   <Route name='app' path='/' handler={App}>
-    <Route name='vote' path='/vote/:electionId' handler={Vote}/>
-    <Route name='submitting' handler={Submitting} />
```
Comment (bportnoy) [inline client/scripts/routes.jsx]: you need to remove this file from your PR - you removed changes that were made in master, if we merge this we'll lose things we've already done.

**Comment by bportnoy** [comment]
Hi all - this isn't quite ready to merge. Closing just in case, we can re-open when Rebekah pushes again.

**Comment by rebekahkwon** [comment]
Cool, thanks for the feedback. I thought I'd push it up as is in case someone wanted to get the needle moving with the routes.

**Comment by rebekahkwon** [comment]
We'll flush more views and clean up with the next pull.

**Pull request closed without merging**